### PR TITLE
Sort dependency project versions in descending order

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary
Sort dependency project versions in descending order

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

demo

### CODE REVIEW

File: packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx [hunk 1]
```diff
@@ -230,7 +230,20 @@ const ProjectDependencyEditor = observer(
     // version
     const version = projectDependency.versionId;
     const versions = selectedProject?.versions ?? [];
-    const versionOptions = versions.map((v) => ({ value: v, label: v }));
+    const versionOptions = versions
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx:233]: could we look into what options we have with `npm`? This doesn't seem like the proper comparison to me. Also, if we cannot go with 3rd party library, perhaps, let's go with an utility function that we can unit test, I would be more confident then.

That being said, why don't we use what the backend return and just revert it, don't they sorta ensure the order for us in some way?

Comment (YannanGao-gs) [reply packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx:233]: this is what backend returns.
Screen Shot 2022-03-04 at 3 35 46 PM
e.g.
1.0 -> 11.2.0 -> 12.0.0 ->  2.1.0

Comment (akphi) [reply packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx:233]: funny how `2.0.0` shows up in the end there.
Anyway, I think we can use this https://www.npmjs.com/package/compare-versions

File: package.json [hunk 2]
```diff
@@ -80,6 +80,9 @@
   "resolutions": {
     "glob-parent": "^6.0.1"
   },
+  "dependencies": {
```
Comment (akphi) [inline package.json:83]: this should be in `@finos/legend-studio`

File: packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx [hunk 3]
```diff
@@ -230,7 +231,9 @@ const ProjectDependencyEditor = observer(
...
+      .map((v) => ({ value: v, label: v }))
+      .sort((v1, v2) => compareVersions(v2.value, v1.value));
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx:236]: probably can do 
```suggestion
      .sort(compareVersions)
      .map((v) => ({ value: v, label: v }))
```

File: packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx (hunk 1 above)
Comment (akphi) [reply packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx:234]: Or we can do this

https://stackoverflow.com/questions/55466274/simplify-semver-version-compare-logic/55466325#55466325

https://github.com/omichelsen/compare-versions/issues/45

That's not bad. We can publish this as part of utilities in `DepotUtils`, we can set a note there linking to this library and this issue.

I can merge this and do that myself if you'd rather me doing that. Let me know!

**Review by akphi — CHANGES_REQUESTED** [review]

File: .changeset/clever-forks-remember.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-server-depot': patch
+---
+
+Add a utility function to compare semver versions.
```
Comment (akphi) [inline .changeset/clever-forks-remember.md:5]:
```suggestion
Add a utility function to compare `semver` versions.
```

**Review by akphi — APPROVED** [review]

**Pull request merged**